### PR TITLE
do not gate backward on requires_grad

### PR DESCRIPTION
**Pull request by chenyuxyz**
DETACH is filtered in _deepwalk. instead of None, it gets 0 grad now

### CODE REVIEW

**Comment by chenyuxyz** [comment]
#16200
